code/modules/mob/living/carbon/xenomorph/castes/praetorian/abilities_praetorian.dm

```diff
@@ -467,3 +467,551 @@ GLOBAL_LIST_INIT(acid_spray_hit, typecacheof(list(/obj/structure/barricade, /obj
+		return
+	if(human_mob.adjustOxyLoss(HUMAN_CRITDRAG_OXYLOSS))
+		return
+	INVOKE_ASYNC(human_mob, TYPE_PROC_REF(/mob/living/carbon/human, adjustBruteLoss), HUMAN_CRITDRAG_OXYLOSS)
+
+
+/// Removes signals for COMSIG_MOVABLE_MOVED and COMSIG_MOVABLE_POST_THROW for thrown humans.
+/datum/action/ability/activable/xeno/dislocate/proc/on_throw_end(datum/source)
+	SIGNAL_HANDLER
+	UnregisterSignal(source, COMSIG_MOVABLE_MOVED)
+	UnregisterSignal(source, COMSIG_MOVABLE_POST_THROW)
+
+// ***************************************
+// *********** Item Throw
+// ***************************************
+/datum/action/ability/activable/xeno/item_throw
+	name = "Item Throw"
+	action_icon_state = "item_throw"
+	action_icon = 'icons/Xeno/actions/praetorian.dmi'
+	desc = "Pick up a nearby item momentarily and throw it in a chosen direction. The item's size determines elements such as how fast or hard it hits."
+	ability_cost = 50
+	cooldown_duration = 12 SECONDS
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_ITEM_THROW,
+	)
+	/// If they have moved at least a single tile since picking up an item. Used as a single-tile grace period before dropping.
+	var/single_tile_grace_period = TRUE
+	/// If we are holding an item.
+	var/obj/item/held_item
+	/// Mutable appearance of the held item.
+	var/mutable_appearance/held_appearance
+
+/datum/action/ability/activable/xeno/item_throw/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return FALSE
+	// Any target is acceptable if we have an item.
+	if(held_item)
+		return TRUE
+	// If we do not, we only want items.
+	if(!isitem(A) || isgrabitem(A))
+		if(!silent)
+			A.balloon_alert(owner, "not an item")
+		return FALSE
+	var/obj/item/item_atom = A
+	if(item_atom.anchored)
+		if(!silent)
+			item_atom.balloon_alert(owner, "item is anchored")
+		return FALSE
+	if(!owner.Adjacent(item_atom))
+		if(!silent)
+			item_atom.balloon_alert(owner, "too far")
+		return FALSE
+
+/datum/action/ability/activable/xeno/item_throw/use_ability(atom/A)
+	if(!held_item)
+		// Grab the floor item like a rouny.
+		playsound(owner, 'sound/voice/alien/pounce2.ogg', 30)
+		var/obj/item/interacted_item = A
+		interacted_item.forceMove(owner)
+		held_item = interacted_item
+		held_appearance = mutable_appearance(interacted_item.icon, interacted_item.icon_state)
+		held_appearance.layer = ABOVE_OBJ_LAYER
+		RegisterSignal(owner, COMSIG_MOVABLE_MOVED, PROC_REF(on_move))
+		RegisterSignal(owner, COMSIG_ATOM_DIR_CHANGE, PROC_REF(owner_turned))
+		owner.add_movespeed_modifier(MOVESPEED_ID_OPPRESSOR_ITEM_GRAB, TRUE, 0, NONE, TRUE, 2)
+		owner_turned(null, null, owner.dir)
+		succeed_activate()
+		ability_cost = 0 // Throwing will cost nothing (and won't cause a warning if not enough plasma).
+		single_tile_grace_period = TRUE
+		return
+	// Throw!
+	owner.remove_movespeed_modifier(MOVESPEED_ID_OPPRESSOR_ITEM_GRAB)
+	held_item.throwforce += held_item.w_class * 15 // Ranges from 15 - 90 damage.
```

Comment:
this can get to absurd numbers. There is NO upper limit to w_class, and we have common w_class modifiers (i.e. a size 4 gun can become size 7, based on attachments).